action.yml

```diff
@@ -58,31 +65,48 @@ outputs:
 runs:
   using: composite
   steps:
+    # We use Cargo throughout the rest of this action for things like generating the lockfile
+    # and tracking the version in the cache key. If it's not installed, exit early. 
+    - name: Check that Cargo is installed
+      shell: bash
+      run: |
+        cargo --version
+
+        # Check that the exit code of the previous command is 0, signaling a success.
+        if [[ $? -eq 0 ]]; then
+          # We are assuming that `rustc` is installed if Cargo is.
+          echo 'Cargo is installed!'
+        else
+          echo 'Cargo is not installed!' >&2
+          echo '  help: Try using the `dtolnay/rust-toolchain` action before this one.' >&2
+          exit 1
+        fi
+
     # The `Cargo.lock` file contains the exact crates used as dependencies.
     # It is important to use for the caching key, because it might become outdated otherwise.
     # If only the `Cargo.toml` file was used, patch releases of crates could cause unnecessary re-compiles.
     # Therefore, if the `Cargo.lock` file does not yet exist (e.g. for libraries), we need to create it first.
     # If the `Cargo.lock` file has been committed to the repo we reuse it to reproduce the same scenario.
     - name: Create Cargo.lock file
       shell: bash
+      env:
+        MANIFEST_PATH: ${{ inputs.manifest-path }}
       run: |
-        if [ -f "Cargo.lock" ]; then
-            echo "Reusing already existing Cargo.lock file."
-        else 
-            echo "Cargo.lock file does not exist, creating it..."
+        # Locate the root `Cargo.toml` from the a given `Cargo.toml` manifest, then use `jq` to
+        # extract the result from JSON.
+        ROOT_MANIFEST=$(cargo locate-project --manifest-path "${MANIFEST_PATH}" --workspace | jq -r '.root')
```

Comment:
Do you know how the `--workspace` option behaves in a non-workspace project?

Reply:
I believe it returns the passed `Cargo.toml`, but let me test it first.

Reply:
Yup, it works! If this command were run in the Bevy repository, it would resolve to `./Cargo.toml` even if `crates/bevy_ecs/Cargo.toml` or `./Cargo.toml` were passed. It even works in non-workspaces.